packages/qwik/src/lib/transition.ts

```diff
@@ -0,0 +1,39 @@
+import type { Transition, TransitionKey } from "@ssgoi/core";
+import { transition as coreTransition } from "@ssgoi/core";
+
+/**
+ * Qwik-specific transition function that properly handles element lifecycle
+ * Usage: Use with useVisibleTask$ to apply transitions
+ *
+ * Example:
+ * const elementRef = useSignal<HTMLElement>();
+ * useVisibleTask$(({ cleanup }) => {
+ *   const element = elementRef.value;
+ *   if (!element) return;
+ *
+ *   const cleanupFn = transition({ key: 'fade', in, out })(element);
+ *   cleanup(() => cleanupFn?.());
+ * });
+ */
+export function transition<TAnimationValue = number>(options: {
+  key: TransitionKey;
+  in?: Transition<undefined, TAnimationValue>["in"];
+  out?: Transition<undefined, TAnimationValue>["out"];
+}) {
+  // Return a function that accepts an element and returns a cleanup function
+  return (element: HTMLElement) => {
+    // Skip if not in browser environment (SSR)
+    if (typeof window === "undefined") return;
```

Comment:
you can import `isBrowser` instead, it's a build time constant and then the code doesn't exist in the server